rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMStatsView.java

```diff
@@ -325,117 +221,193 @@ Canvas build() {
-            controller.setRuntimeRRDRange(r, "AverageActivity");
+            controller.setRuntimeRRDRange(r, AVERAGE_ACTIVITY);
         });
 
         loadHeaderLabel = new Label("<nobr style='font-size:1.4em;font-weight:bold;'>Load History<nobr>");
         loadHeaderLabel.setTooltip("Percentage of cumulative node utilization since server start-up.");
         loadHeaderLabel.setHeight(24);
+    }
 
-        filler = new Canvas();
-        filler.setWidth100();
+    private void initNodeHistoryChart() {
+        /*
+         * Node history graph
+         */
+        nodeHistoryChart = new LineChart();
+        nodeHistoryChart.getOptions().setResponsive(true);
+        nodeHistoryChart.getOptions().getLegend().setDisplay(true);
+        nodeHistoryChart.getOptions().getLegend().setPosition(Position.BOTTOM);
+        nodeHistoryChart.getOptions().getTitle().setDisplay(false);
+        nodeHistoryChart.getOptions().setMaintainAspectRatio(false);
+        nodeHistoryChart.setHeight("250px");
+        nodeHistoryChart.setWidth("100%");
+
+        CartesianLinearAxis yAxis = new CartesianLinearAxis(nodeHistoryChart);
+        yAxis.setDisplay(true);
+        yAxis.getTicks().setBeginAtZero(true);
+        yAxis.getTicks().setStepSize(1);
+        nodeHistoryChart.getOptions().getScales().setYAxes(yAxis);
+
+        CartesianCategoryAxis xAxis = new CartesianCategoryAxis(nodeHistoryChart);
+        xAxis.getGrideLines().setDisplay(false);
+        nodeHistoryChart.getOptions().getScales().setXAxes(xAxis);
+
+        List<Dataset> datasets = new ArrayList<>();
+        for (Map.Entry<String, String> datasetAndColor : datasetsAndColor) {
+            LineDataset dataset = nodeHistoryChart.newDataset();
+            dataset.setFill(Fill.START);
+            dataset.setLabel(datasetAndColor.getKey());
+            dataset.setBorderColor(datasetAndColor.getValue());
+            dataset.setPointBackgroundColor(datasetAndColor.getValue());
+            dataset.setPointBorderColor(datasetAndColor.getValue());
+            dataset.setBackgroundColor(dataset.getBorderColor().alpha(0.2));
+            dataset.setSteppedLine(SteppedLine.AFTER);
+
+            dataset.setPointRadius(2);
+            datasets.add(dataset);
+        }
 
-        HLayout loadHeader = new HLayout();
-        loadHeader.setHeight(24);
-        loadHeader.setMembersMargin(20);
-        loadHeader.setMembers(loadHeaderLabel, filler, loadForm);
+        nodeHistoryChart.getData().setDatasets(datasets.toArray(new Dataset[0]));
 
-        Canvas cc1 = new Canvas();
-        cc1.setHeight(4);
-        cc1.setBackgroundColor("#eee");
-        Canvas cc2 = new Canvas();
-        cc2.setHeight(4);
-        cc2.setBackgroundColor("#eee");
+        nodeLineForm = new DynamicForm();
+        final SelectItem nodeLineSelect = new SelectItem("nodeLineSelect", "");
+        LinkedHashMap<String, String> nodeLineValues = new LinkedHashMap<>();
```

Comment:
You can set new LinkedHashMap size with `new LinkedHashMap<>(StatHistory.Range.values().size())`

Reply:
I don't think that declaring the type LinkHashMap is required. I think that declaring the variable as HashMap is enough and enable to keep a higher level of abstraction.